File: src/tray/createTrayMenu.js
import createTrayIcon from "./createTrayIcon.js";

const icon = await createTrayIcon();

export default function createTrayMenu(updates = []) {
  console.log(updates);
  const count = updates.length;
  return {
    icon: icon,
    tooltip: "Kegger - Join the party",
    items: [
      {
        title: `${count} Updates Available`,
        enabled: false,
      },
      {
        title: "Update All",
        __id: 2,
        enabled: count === 0,
      },
      {
        title: "Clean up Celler",
        __id: 1,
        enabled: true,
      },
      {
        title: "<SEPARATOR>",
      },
      {
        title: "Preferences",
        enabled: true,
        items: [
          {
            title: "Start at login",
            enabled: true,
            checked: true,
          },
        ],
      },
      {
        title: "About",
        enabled: true,
      },
      {
        title: "<SEPARATOR>",
      },
      {
        title: "Quit",
        enabled: true,
        onClick: () => Deno.exit(),
      },
    ],
  };
}

File: cmd/kegger/createTrayMenu.go
package main

import (
	"fyne.io/fyne/v2"
	"github.com/bealearts/kegger/internal/brew"
	"github.com/sqweek/dialog"
)

func createTrayMenu() (*fyne.Menu, *fyne.MenuItem, *fyne.MenuItem) {
	var menu *fyne.Menu
	updatesMenu := fyne.NewMenuItem("0 Updates", func() {})
	updatesMenu.Disabled = true

	updateAllMenu := fyne.NewMenuItem("Update All", func() {
		log.Info("Update All")
		brew.ExecUpdate("")
	})
	updateAllMenu.Disabled = true

	menuItems := []*fyne.MenuItem{
		updatesMenu,
		updateAllMenu,
		fyne.NewMenuItem("Clean up Celler", func() {
			log.Info("Clean up Celler")
			brew.ExecCleanup()
		}),
		fyne.NewMenuItemSeparator(),
		// TODO: Prefs
		fyne.NewMenuItem("About", func() {
			log.Info("About")
			meta := fyne.CurrentApp().Metadata()
			dialog.Message("Kegger - Join the party\n\nVersion %s\nBuild %v", meta.Version, meta.Build).Title("About").Info()
		}),
	}

	menu = fyne.NewMenu("Kegger", menuItems...)

	return menu, updatesMenu, updateAllMenu
}
